supervisor/docker/interface.py

```diff
@@ -150,19 +158,29 @@ def _docker_login(self, image: str) -> None:
 
     @process_lock
     def install(
-        self, version: AwesomeVersion, image: Optional[str] = None, latest: bool = False
+        self,
+        version: AwesomeVersion,
+        image: Optional[str] = None,
+        latest: bool = False,
+        arch: Optional[str] = None,
     ):
         """Pull docker image."""
-        return self.sys_run_in_executor(self._install, version, image, latest)
+        return self.sys_run_in_executor(self._install, version, image, latest, arch)
 
     def _install(
-        self, version: AwesomeVersion, image: Optional[str] = None, latest: bool = False
+        self,
+        version: AwesomeVersion,
+        image: Optional[str] = None,
+        latest: bool = False,
+        arch: Optional[str] = None,
     ) -> None:
         """Pull Docker image.
 
         Need run inside executor.
         """
         image = image or self.image
+        # sys_arch.default not set before plugins and may be an unsupported arch
+        arch = arch or self.sys_supervisor.arch
```

Comment:
I guess we can load it early in the process?

Reply:
I tried that earlier, see the resolved comment chain with myself above. Main problem is [this code](https://github.com/home-assistant/supervisor/blob/main/supervisor/arch.py#L58-L63). When on an unsupported arch sets default arch to that unsupported arch and nothing will have an image for that arch so that will fail.

I realize unsupported architectures are well, unsupported, but that would be pretty catastrophic for anyone in that situation. I'm not sure supervisor would even start anymore since it would just fail to install all the plugins every 30 seconds until shut down.

Reply:
Also I was less sure about this one but it seemed like that might create some edge cases where we picked an image with one arch for supervisor and an image with a different arch for all the plugins. While I don't think that would break anything it did seem odd.

Reply:
Then let's fall back on https://github.com/home-assistant/supervisor/blob/main/supervisor/arch.py#L36 - I created the arch class to have the arch handling central and not refer each time to `self.sys_supervisor.arch` as I did before we had the arch object.